packages/datatrak-web/src/api/mutations/useResubmitSurveyResponse.ts

```diff
@@ -5,113 +5,36 @@
-    }
-
-    if (type === QuestionType.File) {
-      if (isFileUploadAnswer(answer) && answer.value instanceof File) {
-        // Create a new file with a unique name, and add it to the files array, so we can add to the FormData, as this is what the central server expects
-        const uniqueFileName = getUniqueSurveyQuestionFileName(answer.name);
-        files.push(
-          new File([answer.value as Blob], uniqueFileName, {
-            type: answer.value.type,
-          }),
-        );
-        return {
-          ...acc,
-          [code]: uniqueFileName,
-        };
-      }
-      if (answer && typeof answer === 'object' && 'name' in answer) {
-        return {
-          ...acc,
-          [code]: answer.name,
-        };
-      }
-    }
-
-    if (type === QuestionType.DateOfData || type === QuestionType.SubmissionDate) {
-      const date = new Date(answer as string);
-      dataTime = date.toISOString();
-      return acc;
-    }
-    return {
-      ...acc,
-      [code]: answer,
-    };
-  }, {});
-
-  return {
-    answers: formattedAnswers,
-    files,
-    entityId,
-    dataTime,
-  };
-};
+import { AnswersT, useSurveyResponseData } from './useSubmitSurveyResponse';
 
 export const useResubmitSurveyResponse = () => {
   const navigate = useNavigate();
   const params = useParams();
   const { surveyResponseId } = params;
-  const { surveyScreens, resetForm } = useSurveyForm();
-  const allScreenComponents = getAllSurveyComponents(surveyScreens);
-  const questionsById = allScreenComponents.reduce((acc, component) => {
-    return {
-      ...acc,
-      [component.questionId]: component,
-    };
-  }, {});
+
+  const { resetForm } = useSurveyForm();
+
+  const surveyResponseData = useSurveyResponseData();
+  const { data: surveyResponse } = useSurveyResponse(surveyResponseId);
+
   return useMutation<any, Error, AnswersT, unknown>(
-    async (surveyAnswers: AnswersT) => {
-      if (!surveyAnswers) {
+    async (answers: AnswersT) => {
+      if (!answers) {
         return;
       }
-      const { answers, files, entityId, dataTime } = processAnswers(surveyAnswers, questionsById);
-      const timezone = getBrowserTimeZone();
-      const formData = new FormData();
-      const formDataToSubmit = { answers, timezone } as {
-        answers: Record<string, string | number | boolean>;
-        entity_id?: string;
-        data_time?: string;
-        timezone?: string;
-      };
-      if (entityId) {
-        formDataToSubmit.entity_id = entityId;
-      }
-      if (dataTime) {
-        formDataToSubmit.data_time = dataTime;
-      }
-
-      formData.append('payload', JSON.stringify(formDataToSubmit));
-      files.forEach(file => {
-        formData.append(file.name, file);
-      });
 
-      return post(`surveyResponse/${surveyResponseId}/resubmit`, {
-        data: formData,
-        headers: {
-          'Content-Type': 'multipart/form-data',
+      // TODO: check the data time - use existing or now?
```

Comment:
Is this todo intended for now or later?

Reply:
Oh my bad, I actually did address this, just forgot the comment